### PR TITLE
feat(notifications): enable auto page loading and sorting

### PR DESCRIPTION
**Pull request by oyo**
## Description

Enable automatic page loading and sorting for notifications

## Why

After rewrite of the notifications center some features were missing like page loading and ordering. These are added with this PR. Note: the changelog hasn't been updated because the features were existing before and will go into release together with the rewrite PR #407

## Issue

ref: Jira CPLP-3045

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

File: src/components/pages/NotificationCenter/NotificationPager.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/********************************************************************************
+ * Copyright (c) 2021, 2023 BMW Group AG
+ * Copyright (c) 2021, 2023 Contributors to the Eclipse Foundation
```
Comment (evegufy) [inline src/components/pages/NotificationCenter/NotificationPager.tsx:3]: Could you please update the header to '2021, 2024' in every newly added file?
The template for the header still needs to get updated for the new year but I'd do that in a separate PR.

**Review by evegufy — CHANGES_REQUESTED** [review]

File: src/components/pages/NotificationCenter/NotificationPager.tsx (hunk 1 above)
Comment (oyo) [reply src/components/pages/NotificationCenter/NotificationPager.tsx]: Header updated

**Review by evegufy — APPROVED** [review]
